src/components/checkbox/checkbox.ts

```diff
@@ -0,0 +1,246 @@
+  private _currentAnimationClass: string = '';
+
+  private _currentUiState: UiTransitionState = UiTransitionState.Init;
+
+  private _checked: boolean = false;
+
+  private _indeterminate: boolean = false;
+
+  private _changeSubscription: {unsubscribe: () => any} = null;
+
+  constructor(private _renderer: Renderer, private _elementRef: ElementRef) {}
+
+  /**
+   * True when the checkbox is checked, false otherwise. Note that setting `checked` while
+   * will immediately set `indeterminate` to false.
+   */
+  @Input() get checked() {
+    return this._checked;
+  }
+
+  set checked(checked: boolean) {
+    // When checked is set, it immediately overwrites an indeterminate state, regardless of
+    // whether it's true or false.
+    this._indeterminate = false;
+    this._checked = checked;
+    this._transitionUiState(
+        this._checked ? UiTransitionState.Checked : UiTransitionState.Unchecked);
+    this.change.emit(this._checked);
+  }
+
+  /**
+   * True when the checkbox is indeterminate, false otherwise. This is also known as "mixed" mode
+   * and can be used to represent a checkbox with three states, e.g. a checkbox that represents a
+   * nested list of checkable items.
+   */
+  @Input() get indeterminate() {
+    return this._indeterminate;
+  }
+
+  set indeterminate(indeterminate: boolean) {
+    this._indeterminate = indeterminate;
+    if (this._indeterminate) {
+      this._transitionUiState(UiTransitionState.Indeterminate);
+    } else {
+      this._transitionUiState(
+          this.checked ? UiTransitionState.Checked : UiTransitionState.Unchecked);
+    }
+  }
+
+  /** The id that is attached to the checkbox's label. */
+  get labelId() {
+    return `${this.id}-label`;
+  }
+
+  /** Returns the proper aria-checked attribute value based on the checkbox's state. */
+  getAriaChecked() {
+    if (this.indeterminate) {
+      return 'mixed';
+    }
+    return this.checked ? 'true' : 'false';
+  }
+
+  /** Toggles the checked state of the checkbox. If the checkbox is disabled, this does nothing. */
+  toggle(event: Event) {
+    if (this.disabled) {
+      event.stopPropagation();
+      return;
+    }
+    this.checked = !this.checked;
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  writeValue(value: any) {
+    this.checked = !!value;
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  registerOnChange(fn: any) {
+    if (this._changeSubscription) {
+      this._changeSubscription.unsubscribe();
+    }
+    this._changeSubscription = <{unsubscribe: () => any}>this.change.subscribe(fn);
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  registerOnTouched(fn: any) {
+    this.onTouched = fn;
+  }
+
+  private _transitionUiState(newState: UiTransitionState) {
+    const oldState = this._currentUiState;
+    const renderer = this._renderer;
+    const elementRef = this._elementRef;
+
+    if (oldState === newState){
+      return;
+    }
+    if (this._currentAnimationClass) {
```

Comment:
Truthy checks on strings won't work in Dart; conditions have to be exactly `boolean`